lib/controllers/agendash.js

```diff
@@ -56,69 +57,117 @@ module.exports = function (agenda, options) {
-            job: "$$ROOT",
-            _id: "$$ROOT._id",
-            running: {
-              $and: ["$lastRunAt", { $gt: ["$lastRunAt", "$lastFinishedAt"] }],
-            },
-            scheduled: {
-              $and: ["$nextRunAt", { $gte: ["$nextRunAt", new Date()] }],
-            },
-            queued: {
-              $and: [
-                "$nextRunAt",
-                { $gte: [new Date(), "$nextRunAt"] },
-                { $gte: ["$nextRunAt", "$lastFinishedAt"] },
-              ],
-            },
-            completed: {
-              $and: [
-                "$lastFinishedAt",
-                { $gt: ["$lastFinishedAt", "$failedAt"] },
-              ],
-            },
-            failed: {
-              $and: [
-                "$lastFinishedAt",
-                "$failedAt",
-                { $eq: ["$lastFinishedAt", "$failedAt"] },
-              ],
-            },
-            repeating: {
-              $and: ["$repeatInterval", { $ne: ["$repeatInterval", null] }],
-            },
+      },
+      {
+        $project: {
+          job: "$$ROOT",
+          _id: "$$ROOT._id",
+          running: {
+            $and: ["$lastRunAt", { $gt: ["$lastRunAt", "$lastFinishedAt"] }],
+          },
+          scheduled: {
+            $and: ["$nextRunAt", { $gte: ["$nextRunAt", new Date()] }],
+          },
+          queued: {
+            $and: [
+              "$nextRunAt",
+              { $gte: [new Date(), "$nextRunAt"] },
+              { $gte: ["$nextRunAt", "$lastFinishedAt"] },
+            ],
+          },
+          completed: {
+            $and: [
+              "$lastFinishedAt",
+              { $gt: ["$lastFinishedAt", "$failedAt"] },
+            ],
+          },
+          failed: {
+            $and: [
+              "$lastFinishedAt",
+              "$failedAt",
+              { $eq: ["$lastFinishedAt", "$failedAt"] },
+            ],
+          },
+          repeating: {
+            $and: ["$repeatInterval", { $ne: ["$repeatInterval", null] }],
           },
         },
-        { $match: postMatch },
-        {
-          $facet: {
-            pages: [
-              { $count: "totalMatchs" },
-              {
-                $project: {
-                  totalPages: {
-                    $ceil: { $divide: ["$totalMatchs", options.limit] },
-                  },
+      },
+      { $match: postMatch }
+
+      
+    ];
+    const standardMongoDBQuery = [
+
+      {
+        $facet: {
+          pages: [
+            { $count: "totalMatchs" },
+            {
+              $project: {
+                totalPages: {
+                  $ceil: { $divide: ["$totalMatchs", options.limit] },
                 },
               },
-            ],
-            filtered: [{ $skip: options.skip }, { $limit: options.limit }],
+            },
+          ],
+          filtered: [{ $skip: options.skip }, { $limit: options.limit }],
+        }
+      }
+    ]
+    const awsDocumentDBQuery = [
```

Comment:
Can we explain the limitation with this query, for some reason we're trying to run a query, failing and then trying to run another query that's docDB compatible.